lib/measured/unit_system_builder.rb

```diff
@@ -8,12 +8,56 @@ def unit(unit_name, aliases: [], value: nil)
     nil
   end
 
+  def si_unit(unit_name, aliases: [], value: nil)
+    @units += build_si_units(unit_name, aliases: aliases, value: value)
+    nil
+  end
+
   def build
     Measured::UnitSystem.new(@units)
   end
 
   private
 
+  def self.prefixes
+    [
+      ['y', 'yocto', -24],
+      ['z', 'zepto', -21],
+      ['a', 'atto', -18],
+      ['f', 'femto', -15],
+      ['p', 'pico', -12],
+      ['n', 'nano', -9],
+      ['μ', 'micro', -6],
+      ['m', 'milli', -3],
+      ['c', 'centi', -2],
+      ['d', 'deci', -1],
+      ['', '', 0],
+      ['da', 'deca', 1],
+      ['h', 'hecto', 2],
+      ['k', 'kilo', 3],
+      ['M', 'mega', 6],
+      ['G', 'giga', 9],
+      ['T', 'tera', 12],
+      ['P', 'peta', 15],
+      ['E', 'exa', 18],
+      ['Z', 'zetta', 21],
+      ['Y', 'yotta', 24]
+    ]
+  end
+
+  def build_si_units(name, aliases: [], value: nil)
+    value = "1 #{name}" if !value
+    size, unit = value.split(' ')
+    size ||= 1
```

Comment:
So I'd rather avoid doing the `split` above. If we absolutely must, it's important to share the code that deals with this, specifically [`Unit#parse_value`](https://github.com/Shopify/measured/blob/58470f5cb085384d1d77484aeb9f21e9b7de0456/lib/measured/unit.rb#L66-L77).

What I'd actually recommend though, is special casing the base unit's value, so you start off with something like
```ruby
si_units = [build_unit(name, aliases: aliases, value: value)]
```

The cool "trick" you can use after that is that all of the prefix units can then have conversion values based on the base unit! So you could do this in your loop below:

```ruby
si_units << build_unit("#{short}#{name}", aliases: long_names, value: "10e#{exp} #{name}")
```

This means you don't have to unpack value. It's also important for a special case: what if `si_unit` is the base unit of our entire system, upon which all other conversions are based? In that case `value` is `nil` and things wouldn't work (as they are now).